chain/actions/src/payload.rs

```diff
@@ -87,58 +93,47 @@ pub trait PayloadGenerator<T: Into<TypedTransaction>> {
 }
 
 fn channels_payload(hopr_channels: Address, call_data: Vec<u8>) -> Vec<u8> {
-    ExecTransactionFromModuleCall {
+    execTransactionFromModuleCall {
         to: hopr_channels.into(),
-        value: U256::zero(),
+        value: U256::ZERO,
         data: call_data.into(),
         operation: Operation::Call as u8,
     }
-    .encode()
+    .abi_encode()
 }
 
-fn approve_tx(spender: Address, amount: Balance) -> TypedTransaction {
-    let mut tx = create_eip1559_transaction();
-    tx.set_data(
-        ApproveCall {
+fn approve_tx(spender: Address, amount: Balance) -> TransactionRequest {
+    TransactionRequest::default().with_input(
+        approveCall {
             spender: spender.into(),
-            value: amount.amount(),
+            value: U256::from_be_bytes(amount.amount().to_be_bytes()),
```

Comment:
Isn't `alloy::primitives::U256` compatible with `primitive_types::U256` - which is our `hopr_primitive_types::prelude::U256` ?

Reply:
Yes. I added the From Implementation but sometimes it's not recognized...cargo check blames a lot - I wonder why?

Reply:
alloy uses `ruint` and we use `primitive_types`. I guess that hinders direct conversion?

Reply:
Ok, maybe we should also start using `ruint` in `hopr-primitive-types` as our `U256` then.

Reply:
I'll create an issue for it and keep the conversion like this for the moment in the codebase. WDYT?

Reply:
Sounds good.